### PR TITLE
raft_client: connect to other stores using peer_address

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

###  What have you changed?
We plan to provide a heterogeneous engine with tikv, which may need to use different addresses to provide kv and raft services. Now kvproto has an optional `peer_address` to support this. 

The corresponding tikv modification is: use `peer_address` when connecting other nodes to send raft messages. (for backward compatibility: if `peer_address` is not set, `address` is still used)

###  What is the type of the changes?
- New feature (a change which adds functionality)

###  How is the PR tested?
- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
No

###  Refer to a related PR or issue link (optional)
kvproto commits: https://github.com/pingcap/kvproto/pull/462 https://github.com/pingcap/kvproto/pull/464

### CODE REVIEW

**Comment by siddontang** [comment]
em, I don't like the name of engine_address, why not name peer_addr like etcd?

**Comment by disksing** [comment]
@siddontang ok. I'll update kvproto first.

**Comment by disksing** [comment]
ping @overvenus @siddontang

File: src/server/resolve.rs [hunk 1]
```diff
@@ -69,14 +69,14 @@ impl<T: PdClient> Runner<T> {
...
-        let mut s = box_try!(pd_client.get_store(store_id));
+        let s = box_try!(pd_client.get_store(store_id));
         if s.get_state() == metapb::StoreState::Tombstone {
             RESOLVE_STORE_COUNTER
                 .with_label_values(&["tombstone"])
                 .inc();
             return Err(box_err!("store {} has been removed", store_id));
         }
-        let addr = s.take_address();
+        let addr = get_peer_address(&s);
```
Comment (siddontang) [inline src/server/resolve.rs]: can we take the address here?

Comment (disksing) [reply src/server/resolve.rs]: looks doable.

**Comment by disksing** [comment]
Please take another look @siddontang @BusyJay

**Comment by disksing** [comment]
fail due to https://github.com/tikv/tikv/issues/5628
rerun
/run-test

**Pull request merged**